src/ReachSets/discretize.jl

```diff
@@ -647,6 +682,45 @@ function discretize_interpolation(𝑆::InitialValueProblem{<:AbstractContinuous
     else
         throw(ArgumentError("input of type $(typeof(U)) is not allowed"))
     end
+    return Ω0, Ud
+end
 
-    return IVP(CLCDS(ϕ, Id(size(A, 1)), nothing, Ud), Ω0)
+# version using concrete operations with zonotopes
+function _discretize_interpolation_homog(X0, ϕ, Einit, set_operations::Val{:zonotope})
+    Einit = convert(Zonotope, Einit)
+    Z1 = convert(Zonotope, X0)
+    Z2 = linear_map(ϕ, minkowski_sum(X0, Einit))
+    Ω0 = overapproximate(ConvexHull(Z1, Z2), Zonotope)
+    return Ω0
+end
+
+# version using concrete operations with zonotopes
+function _discretize_interpolation_inhomog(δ, U0, U, X0, ϕ, Einit, Eψ0, A, sih, Phi2Aabs, set_operations::Val{:zonotope})
+    Einit = convert(Zonotope, Einit)
+    Eψ0 = convert(Zonotope, Eψ0)
+    Z1 = X0
+    ϕX0 = linear_map(ϕ, X0)
+    U0 = convert(Zonotope, U0)
+    δI = Matrix(δ*I, size(U0))
```

Comment:
What is `size(U0)`? `U0` is a `Zonotope`.

Reply:
oops, this should be `dim(U0)`, thanks

Reply:
```julia
julia> using LinearAlgebra

julia> δ = 0.1;

julia> Matrix(δ*I, 3)
ERROR: MethodError: no method matching Array{T,2} where T(::UniformScaling{Float64}, ::Int64)
```

Reply:
`Matrix` requires both number of rows and number of columns; `size` gives a tuple, so the correct usage is `Matrix(δ*I, dim(U0), dim(U0))`